apps/login/ui/SignInWithIDP.tsx

```diff
@@ -0,0 +1,132 @@
+
+export function SignInWithIDP({
+  host,
+  identityProviders,
+  startIDPFlowPath = START_IDP_FLOW_PATH,
+}: SignInWithIDPProps) {
+  const [loading, setLoading] = useState<boolean>(false);
+  const [error, setError] = useState<string>("");
+  const router = useRouter();
+
+  async function startFlow(idpId: string, provider: ProviderSlug) {
+    setLoading(true);
+
+    const res = await fetch("/api/idp/start", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        idpId,
+        successUrl: `${host}/register/idp/${provider}/success`,
+        failureUrl: `${host}/register/idp/${provider}/failure`,
+      }),
+    });
+
+    const response = await res.json();
+
+    setLoading(false);
+    if (!res.ok) {
+      setError(response.details);
+      return Promise.reject(response.details);
+    }
+    return response;
+  }
+
+  return (
+    <div className="flex flex-col w-full space-y-2 text-sm">
+      {identityProviders &&
+        identityProviders.map((idp, i) => {
+          switch (idp.type) {
+            case 6: // IdentityProviderType.IDENTITY_PROVIDER_TYPE_GITHUB:
+              return (
+                <SignInWithGithub
+                  key={`idp-${i}`}
+                  onClick={() =>
+                    startFlow(idp.id, ProviderSlug.GITHUB).then(
+                      ({ authUrl }) => {
+                        router.push(authUrl);
+                      }
+                    )
+                  }
+                ></SignInWithGithub>
+              );
+            case 7: // IdentityProviderType.IDENTITY_PROVIDER_TYPE_GITHUB_ES:
+              return (
+                <SignInWithGithub
+                  key={`idp-${i}`}
+                  onClick={() => alert("TODO: unimplemented")}
+                ></SignInWithGithub>
+              );
+            case 5: // IdentityProviderType.IDENTITY_PROVIDER_TYPE_AZURE_AD:
+              return (
+                <SignInWithAzureAD
+                  key={`idp-${i}`}
+                  onClick={() => alert("TODO: unimplemented")}
+                ></SignInWithAzureAD>
+              );
+            case 10: // IdentityProviderType.IDENTITY_PROVIDER_TYPE_GOOGLE:
+              return (
+                <SignInWithGoogle
+                  key={`idp-${i}`}
+                  e2e="google"
+                  name={idp.name}
+                  onClick={() =>
+                    startFlow(idp.id, ProviderSlug.GOOGLE).then(
+                      ({ authUrl }) => {
+                        router.push(authUrl);
+                      }
+                    )
+                  }
+                ></SignInWithGoogle>
+              );
+            case 8: // IdentityProviderType.IDENTITY_PROVIDER_TYPE_GITLAB:
+              return (
+                <SignInWithGitlab
+                  key={`idp-${i}`}
+                  onClick={() => alert("TODO: unimplemented")}
```

Comment:
Why are not all providers implemented?

Reply:
This can be done in a follow up but for this user story it is out of the scope, IMHO

Reply:
Why? Is that discussed with @hifabienne? Where is the follow-up?

Reply:
We will never be able to have ALL providers implemented and its not part of the user story.

Reply:
If we need examples for ALL providers we should cover that seperately and @hifabienne should create the user story for it

Reply:
I am not sure what it is about, in the sdk we should cover all the providers. Also in our hosted login we should cover them, otherwise we would not be able to show them if a customer configures it on the settings

Reply:
Ok so it is like this:

The SDK needs to have the IDPs implemented once we release it properly to the "public".

Short-term goal is though to (a) verify that our approach is working and (b) to improve the signup of our cloud service.
That's why we (@peintnermax and I) opted to use Google, GitHub as initial two providers before implementing all the others.

We can and should do a follow-up to implement all other providers provided by the API as a separate task, or does somebody disagree with this?

Reply:
No thats fine by me. So either we do not close the issue as it is not finished or we need some follow up issue. @peintnermax can you please do that as you know what is done and what is missing?

Reply:
I'll create an other user story